src/main/java/io/codelex/flightplanner/repository/FlightDbRepository.java

```diff
@@ -0,0 +1,34 @@
+package io.codelex.flightplanner.repository;
+
+import io.codelex.flightplanner.domain.Airport;
+import io.codelex.flightplanner.domain.Flight;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public interface FlightDbRepository extends JpaRepository<Flight, Integer> {
+
+
+    @Query("SELECT DISTINCT f.from FROM Flight f WHERE LOWER(f.from.airport) LIKE CONCAT('%', :phrase, '%')\n" +
+            "                                       OR LOWER(f.from.city) LIKE CONCAT('%', :phrase, '%')\n" +
+            "                                       OR LOWER(f.from.country) LIKE CONCAT('%', :phrase, '%')")
+    List<Airport> searchAirports(@Param("phrase") String phrase);

```

Comment:
Interesanti - kāpēc lidostas nemeklēji ar Airport repository palīdzību, bet gan šeit?
Mīnuss šim variantam ir tāds, ka viņš nekad neatradīs lidostas, ja tā lidosta ir tikai TO laukā nevis FROM. Citiem vārdiem - šobrīd Tu meklē lidostās, kuras ir bijušas FROM laukā.